### PR TITLE
v2 template endpoint

### PR DESCRIPTION
**Pull request by leohemsted**
feedback welcome i'm very lost in all this

### CODE REVIEW

File: spec/integration/ClientSpec.php [hunk 1]
```diff
@@ -29,7 +29,6 @@ function let(){
 
         $this->beConstructedWith([
             'baseUrl'       => getenv('NOTIFY_API_URL'),
-            'serviceId'     => getenv('SERVICE_ID'),
```
Comment (afoldesi-gds) [inline spec/integration/ClientSpec.php]: Could be left in to ensure backwards compatibility - we haven't yet migrated over... :(

File: spec/unit/ClientSpec.php [hunk 2]
```diff
@@ -626,4 +626,184 @@ function it_receives_an_exception_when_the_api_returns_500(){
...
 
+    //----------------------------------------------------------------------------------------------------------
+    // Template lookups (GETs) with expected success
+
+    function it_generates_the_expected_request_when_looking_up_a_template(){
+
+        //---------------------------------
+        // Test Setup
+
+        $id = '35836a9e-5a97-4d99-8309-0c5a2c3dbc72';
```
Comment (afoldesi-gds) [inline spec/unit/ClientSpec.php]: From ENV?

File: spec/integration/ClientSpec.php (hunk 1 above)
Comment (leohemsted) [inline spec/integration/ClientSpec.php]: hey - this is just our integration tests that we run on code change, which use the new format - don't worry, we're not invalidating the old version just yet 🙂

File: spec/unit/ClientSpec.php (hunk 2 above)
Comment (leohemsted) [inline spec/unit/ClientSpec.php]: this is just a random ID i copied from further up the page, and [hopefully] holds no relevance in the real world - just used it for this unit test to prove it gets passed through into the URL

Comment (afoldesi-gds) [inline spec/unit/ClientSpec.php]: *Copied to further down :D - how a constant?

Comment (leohemsted) [inline spec/unit/ClientSpec.php]: hah perhaps that's not a bad idea 😉

File: spec/integration/ClientSpec.php (hunk 1 above)
Comment (afoldesi-gds) [inline spec/integration/ClientSpec.php]: Thanks, just wanted to make sure that it's tested even if the param is optional - found the unit test ;)

**Comment by leohemsted** [comment]
thanks for your input @afoldesi-gds 😄

File: spec/integration/ClientSpec.php [hunk 3]
```diff
@@ -287,4 +286,129 @@ function it_receives_the_expected_response_when_looking_up_all_notifications() {
...
+      $response->shouldHaveKey( 'body' );
+      $response->shouldHaveKey( 'subject' );
+
+      $response['id']->shouldBeString();
+      $response['id']->shouldBe( $templateId );
+      $response['type']->shouldBeString();
+      $response['type']->shouldBe( 'sms' );
+      $response['version']->shouldBeInteger();
+      $response['body']->shouldBe("Hello ((name))\n\nFunctional Tests make our world a better place");
+      $response['subject']->shouldBeNull();
```
Comment (imdadahad) [inline spec/integration/ClientSpec.php]: Do we also want to add another test like this for email so that ensure subject has the right value?

**Pull request merged**